module/enrichers.mjs

```diff
@@ -823,6 +903,36 @@ async function rollAction(event) {
         case "tool":
           options.ability = ability;
           return await actor.rollToolCheck(tool, options);
+        case "item":
+          if ( target.dataset.rollItemUuid ) {
+            const gameActor = await fromUuid(target.dataset.rollItemActor);
+            if ( gameActor.testUserPermission(game.user, "OWNER") ) {
+              return (await fromUuid(target.dataset.rollItemUuid)).use();
+            } else {
+              return ui.notifications.warn(`You do not have ownership of ${gameActor?.name}, and cannot use this item.`);
+            }
+          } else if ( target.dataset.rollItemName ) {
+            if ( target.dataset.rollItemActor ) {
+              const gameActor = await fromUuid(target.dataset.rollItemActor);
+              if ( gameActor.testUserPermission(game.user, "OWNER") ) {
```

Comment:
When we just check for the current user, this can be simplified:
```suggestion
              if ( gameActor.isOwner ) {
```
This (and some other lines of code) is repeated elsewhere; it would probably be good to move stuff further up the chain and just call them once, like using `fromUuid` up at the top just once.